app/static/src/app/components/code/CodeEditor.vue

```diff
@@ -29,23 +29,92 @@ export default defineComponent({
         const store = useStore();
 
         const editor = ref<null | HTMLElement>(null); // Picks up the element with 'plot' ref in the template
+        const oldDecorations = ref<string[]>([]);
 
         const currentCode = computed(() => store.state.code.currentCode);
         const readOnly = computed(() => (store.state.config as AppConfig).readOnlyCode);
         const defaultCode = computed(() => (store.state.config as AppConfig).defaultCode);
         const defaultCodeExists = computed(() => defaultCode.value && (defaultCode.value.length && !readOnly.value));
+        const odinModelResponse = computed(() => store.state.model.odinModelResponse as OdinModelResponse);
 
         let newCode: string[] | null = null;
         let timeoutId: null | Timeout = null;
         let editorInstance: monaco.editor.IStandaloneCodeEditor;
-        const updateCode = () => store.dispatch(`code/${CodeAction.UpdateCode}`, newCode, { root: true });
+
+        const getMonacoRange = (line: number, reset = false) => {
+            return {
+                startLineNumber: line,
+                startColumn: reset ? 1 : 0,
+                endLineNumber: line,
+                endColumn: reset ? 1 : Infinity
+            };
+        };
+
+        type EditorStates = "error" | "warning"
+
+        type EditorGlyphs = {
+            [K in EditorStates]: string
+        }
```

Comment:
I'd be inclined to pull types outside the `setup` method to declutter it a bit

Reply:
done